### PR TITLE
centos7 compatibility

### PR DESCRIPTION
**Pull request by danykaufman**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3,12 +3,11 @@
 def buildContainer(distro, tag) {
     docker_file = "Dockerfile_${distro}"
     image_name = "devops/${distro}_kubespray_builder:${tag}"
-    common.shell(['docker', 'build', '--no-cache', '-t', image_name, '-f', docker_file, '.'])
+    common.shell(['docker', 'build', '-t', image_name, '-f', docker_file, '.'])
```
Comment (danykaufman) [inline Jenkinsfile:6]: Replacing COPY with ADD in the Dockerfile lets us use caching

File: Jenkinsfile [hunk 2]
```diff
@@ -31,7 +30,7 @@ common.main {
         timestamps {
 
             stage('checkout') {
-                deleteDir()
+                sh "sudo rm -rf ${WORKSPACE}/*"
```
Comment (danykaufman) [inline Jenkinsfile:33]: Fixes deleteDir() root permissions issue

File: igz_files/igz_make_offline.sh [hunk 3]
```diff
@@ -4,10 +4,16 @@
 set -e
 . config.sh
 
-CURRENT_DIR=$(pwd)
 KUBESPRAY_DIR=./cache/kubespray-${KUBESPRAY_VERSION}
 # Current dir already has a slash at the end - don't fix the line below
-PATCH_DIR=${CURRENT_DIR}target-scripts/patches/${KUBESPRAY_VERSION}
+PATCH_DIR=target-scripts/patches/${KUBESPRAY_VERSION}
+
+# Apply pre-install patches
```
Comment (danykaufman) [inline igz_files/igz_make_offline.sh:11]: Needed to patch pre-kubespray shell scripts

File: igz_files/igz_make_offline.sh [hunk 4]
```diff
@@ -4,10 +4,16 @@
 set -e
 . config.sh
 
-CURRENT_DIR=$(pwd)
```
Comment (danykaufman) [inline igz_files/igz_make_offline.sh:7]: used only once and defaults to / so not really needed

File: igz_files/igz_make_offline.sh [hunk 5]
```diff
@@ -22,33 +28,31 @@ cp $KUBESPRAY_DIR/requirements.txt .
 ls -la .
 
 # We need to break the flow to set kube_version and apply the required patches
-
-if [[ ! -d "${PATCH_DIR}" ]]; then
-  mkdir -p ${PATCH_DIR}
-fi
-
 if [[ -d "./igz_patches/${KUBESPRAY_VERSION}" ]]; then
+  mkdir -p ${PATCH_DIR}
```
Comment (danykaufman) [inline igz_files/igz_make_offline.sh:32]: create dir only if required

File: igz_files/igz_make_offline.sh [hunk 6]
```diff
@@ -22,33 +28,31 @@ cp $KUBESPRAY_DIR/requirements.txt .
...
+    else
+      echo "Found patch file $file"
+      patches+=("$file")
+    fi
+  done
+fi
 
 # We apply the patches here mostly because of kube_version.patch otherwise the script will work on the default k8s version for current release
 # Apply all patches
-for patch in "${patchArray[@]}"; do
```
Comment (danykaufman) [inline igz_files/igz_make_offline.sh:51]: this never works - patchArray is never defined

File: igz_build_ansible.sh [hunk 7]
```diff
@@ -4,6 +4,13 @@ source config.sh
 source scripts/common.sh
 source scripts/images.sh
 
+# Apply post-install patches
```
Comment (danykaufman) [inline igz_build_ansible.sh:7]: needed to patch files outside of the containers

File: Jenkinsfile (hunk 2 above)
Comment (alexpirogovski) [reply Jenkinsfile:33]: Will it work on all builders?

**Review by alexpirogovski — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (danykaufman) [reply Jenkinsfile:33]: worked on all the builders i used today ;-)  (a lot)

**Pull request merged**